### PR TITLE
(Wallet) Implement new password creation screen

### PR DESCRIPTION
**Pull request by simoarpe**
## Resolves https://github.com/brave/brave-browser/issues/36552

Implements new password creation screen following [Figma design](https://www.figma.com/file/oXIiv45pIhF2RGlpEjOfUg/Wallet-onboarding?node-id=6223%3A70942&mode=dev).

Main umbrella issue: https://github.com/brave/brave-browser/issues/35294

https://github.com/brave/brave-core/assets/3308503/e6d7fd79-9aec-48eb-bb40-12e43a57ebda

### Notable Changes

- Implements responsive layout that adjust continue button position with and without software keyboard.
- Implements stronger checks that prevents multiple tappings on the buttons loading the same fragment twice.
- Moves wallet creation logic into appropriate fragment `OnboardingCreatingWalletFragment`.
- Implements new UI component `PasswordStrengthMeterView` following latest Material guidelines.
- Migrate away from deprecated `FragmentStatePagerAdapter` in favor of newer `FragmentStateAdapter` and `ViewPager2` bringing better perfomance and removing callback duplication.

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Create a new Wallet
- Accept terms of use
- Enter a password in the first edit text
- Re-type the password in the second edit text
- Observe the match sign pops up and the button becomes enabled
- Modify the password in the first edit text
- Observe the button becames disabled
- Increase password length
- Observe weak, medium, and strong match the logic of Desktop
- Try to tap on continue button multiple times
- Observe the button can be tapped only when the passwords match and only a single time

### CODE REVIEW

**Comment by thypon** [comment]
@simoarpe is this Android only?

File: android/java/org/chromium/chrome/browser/crypto_wallet/adapters/CryptoWalletOnboardingPagerAdapter.java [hunk 1]
```diff
@@ -6,17 +6,16 @@
...
-import androidx.fragment.app.FragmentStatePagerAdapter;
+import androidx.fragment.app.FragmentActivity;
+import androidx.viewpager2.adapter.FragmentStateAdapter;
 
 import org.chromium.chrome.browser.crypto_wallet.util.NavigationItem;
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class CryptoWalletOnboardingPagerAdapter extends FragmentStatePagerAdapter {
```
Comment (simoarpe) [inline android/java/org/chromium/chrome/browser/crypto_wallet/adapters/CryptoWalletOnboardingPagerAdapter.java:19]: Migrated away from deprecated `FragmentStatePagerAdapter`.

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Review by mkarolin — APPROVED** [review]
`strings`++

**Comment by diracdeltas** [comment]
Tagging @stoletheminerals for security review since this is a security-related change.

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding/OnboardingSecurePasswordFragment.java [hunk 2]
```diff
@@ -10,161 +10,124 @@
...
+                            BiometricPrompt.AuthenticationResult result) {
+                        super.onAuthenticationSucceeded(result);
+                        KeystoreHelper.useBiometricOnUnlock(password);
+                        goToTheNextPage(password);
+                    }
+
+                    @Override
+                    public void onAuthenticationError(int errorCode, CharSequence errString) {
+                        super.onAuthenticationError(errorCode, errString);
+                        // Even though we have an error, we still let to proceed
```
Comment (stoletheminerals) [inline android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding/OnboardingSecurePasswordFragment.java:106]: I know it's a new code, but can you explain why do we still proceed here?

**Review by stoletheminerals — APPROVED** [review]

**Pull request merged**